### ISSUE TITLE
OpenCAS Metadata ERROR at Rebooting

### ISSUE THREAD

**Issue by ycs0220** [Open-CAS/open-cas-linux#455]
### Environment
- OS: RedHat-7.5
- kernel: 3.10.0-862.11.6.el7.x86_64
- OpenCAS: OpenCAS-20.3
### issue
when i reboot system, I will see Metadata ERROR information in /var/log/messages。the information is as follows
![image](https://user-images.githubusercontent.com/34889631/86340584-b581a200-bc87-11ea-9006-82db798a1e69.png)

**Comment by eriksun421** [comment]
Hi, can you please provide following information ?
1. CAS configuration: /etc/opencas/opencas.conf
2. casadm -L
3. casadm -P -i X    // X is your CAS instance ID you assigned when it is created. 
4. Detailed steps to reproduce this issue ?  Is it a dirty shutdown or clean shutdown ?  Is there any filesystem or LVM on top of CAS device ?  Is there any I/O requests on CAS device when you reboot the system ? Is it 100% reproducible ? This is very important information that can help us analyze this issue. Please provide as much details as you can.

**Comment by ycs0220** [comment]
> Hi, can you please provide following information ?
> 
> 1. CAS configuration: /etc/opencas/opencas.conf
> 2. casadm -L
> 3. casadm -P -i X    // X is your CAS instance ID you assigned when it is created.
> 4. Detailed steps to reproduce this issue ?  Is it a dirty shutdown or clean shutdown ?  Is there any filesystem or LVM on top of CAS device ?  Is there any I/O requests on CAS device when you reboot the system ? Is it 100% reproducible ? This is very important information that can help us analyze this issue. Please provide as much details as you can.

- - -
Thank you for your reply! Some detailes  as follows:

- /etc/opencas/opencas.conf
```
[caches]
1	/dev/qdisk/P0B00S07	WB	cache_line_size=8,cleaning_policy=acp
2	/dev/qdisk/P0B00S08	WB	cache_line_size=8,cleaning_policy=acp
[cores]
1	1	/dev/qdisk/P0B00S05
2	1	/dev/qdisk/P0B00S06
```
  
- casadm -L
```
#casadm -L
type    id   disk       status    write policy   device
cache   1    /dev/sdb   Running   wb             -
└core   1    /dev/sdi   Active    -              /dev/cas1-1
cache   2    /dev/sdc   Running   wb             -
└core   1    /dev/sdj   Active    -              /dev/cas2-1
```
  
- casadm -P -i x
![image](https://user-images.githubusercontent.com/34889631/86426931-80bf2a80-bd1b-11ea-915f-bc4e9a3d2cbf.png)

  
There is no any filesystem or LVM on top of CAS device, we do two cases to Reproduce this problem, Metadata ERROR occurs in both cases

- There are any I/O requests on CAS device when i reboot the system
- There are no I/O requests on CAS device when i reboot the system

**Comment by imjfckm** [comment]
Hi @ycs0220,

I have a few questions for you to help us pin down the issue you're observing.
- What is your reproduction rate?
- Does this reproduce each time you reboot?
- Are there any other logs while rebooting which would indicate some kind of I/O errors?
- Is there any chance that there is some other piece of software writing to cache device, not only CAS?

This section of metadata is extremely unlikely to produce such errors consistenly, especially without any I/Os while rebooting... So there must be something strange going on.

**Comment by ycs0220** [comment]
This problem was fixed in version V20.3.1, i will close this issue.

**Comment by ycs0220** [comment]
@robertbaldyga @imjfckm @eriksun421 @argretzi 
Thanks for your reply!
This problem was fixed in version V20.3.1, i will close this issue.
This is related to the server S3/S4 status suggested in the previous intelcas document.
